controller/share_manager_controller.go

```diff
@@ -480,23 +476,26 @@ func (c *ShareManagerController) syncShareManagerVolume(sm *longhorn.ShareManage
 		sm.Status.State = types.ShareManagerStateStarting
 	}
 
-	if volume.Spec.NodeID != sm.Status.OwnerID || volume.Status.CurrentNodeID != sm.Status.OwnerID {
```

Comment:
I am not sure if this part can be replaced. Since we want to make sure sm.Status.OwnerID == volume.Spec.NodeID == volume.Status.CurrentNodeID == pod.Spec.NodeName in this controller. Checking the node down only may be not enough.

Reply:
IMHO,  we only need `volume.Spec.NodeID` == `pod.Spec.NodeName` so that the share manager pod can access the underlying Longhorn volume on the same node. The `sm.Status.OwnerID` and `volume.Status.CurrentNodeID` should be designed so that they can be moved freely to any controller node without disrupting the current running workloads that use the RWX volume. How do you think?
cc @joshimoo

Reply:
I am confused about the share manager state and the node ID issue now:
1. Can `sm.status.ownerID` be different from `volume.Spec.NodeID == pod.Spec.NodeName` while the related RWX volume is running?
2. Then when will the share manager become state unknown?

Reply:
> Can sm.status.ownerID be different from volume.Spec.NodeID == pod.Spec.NodeName while the related RWX volume is running?

Yes, it can from the code perspective. I have also tested the scenario 

> Then when will the share manager become state unknown?

It never becomes unknown from the code perspective.  We also mark it as `error` when things went wrong. I think it is ok.

Reply:
Using status.Owner as condition was a simplification for volume failure handling.
It's not strictly required that the owner for the crd is the same node as the actual pod.
So this change is fine, but comes with the caveat that it adds a couple more isNodeDown checks  :)

But overall I agree, it's okay to decouple CRD processing from pod/volume ownership.